### PR TITLE
Fix: bugs for huggingface runtime template

### PR DESCRIPTION
**Pull request by yuzisun**
**What this PR does / why we need it**:
- Remove model id and model dir from template as they can't both exist
- Add option to return token type id
 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

1.  The current template does not work for the case when `storage_uri` is provided 

2. Tested a T5 model which returns following error
```
  File "/Users/dsun20/go/github.com/kserve/python/huggingfaceserver/huggingfaceserver/model.py", line 235, in predict
    raise InferenceError(str(e))
kserve.errors.InferenceError: The following `model_kwargs` are not used by the model: ['token_type_ids'] (note: typos in the generate arguments will also show up in this list)
```

**Type of changes**
Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/controller/v1beta1/inferenceservice/utils/utils.go [hunk 1]
```diff
@@ -64,6 +64,12 @@ func IsMMSPredictor(predictor *v1beta1api.PredictorSpec) bool {
 		}
 		return false
 	} else {
+		if predictor.Model != nil {
+			// we do not support model agent for Huggingface runtime
+			if predictor.Model.ModelFormat.Name == "huggingface" {
+				return false
+			}
+		}
```
Comment (terrytangyuan) [inline pkg/controller/v1beta1/inferenceservice/utils/utils.go]: Oops you are faster. I am fixing this in https://github.com/kserve/kserve/pull/3449 that covers one more scenario as well as test case. Would you like to merge that first and then rebase?

Comment (yuzisun) [reply pkg/controller/v1beta1/inferenceservice/utils/utils.go]: sounds good

Comment (yuzisun) [reply pkg/controller/v1beta1/inferenceservice/utils/utils.go]: My PR has a couple of other bug fixes

**Review by terrytangyuan — APPROVED** [review]
/lgtm

**Pull request merged**